### ISSUE TITLE
open3d_slam_ros inconclusive (rgb-d camera)

### ISSUE THREAD

**Issue by enginksz** [leggedrobotics/open3d_slam#40]
Hello,

I have completed the installation with the mentioned steps.

roslaunch open3d_slam_ros mapping.launch

command or launch files in open3d_slam_ros output only using lidar sensor?

can i get results with stereo or rgb-d camera in my hand? If so, can you specify how to do it?

Thanks

**Comment by enginksz** [comment]
Yes, I want to use rgb-d camera instead of lidar and create/save a point cloud. I used an rgb-d camera but I couldn't see a result on the rviz screen.

**Comment by jelavice** [comment]
Hi enginksz, 

You need to change the topic name to correspond to the one your depth camera publishes. You also need to change the tuning for the depth camera since lidar and depth cameras are very different sensors. Inside the param folder, you will find the params_realsense.yaml file which we used for mapping with real sense depth camera. You can change the path to the config file in the launch file. Bear in mind that you need to avoid fast motions and large open spaces with no geometry, since the depth camera since it has a very limited FOV compared to the lidar. The pointcloud should be saved at the end automatically, if not make sure it is enabled in the configuration *.yaml file.

**Comment by enginksz** [comment]
Hello,

I changed the launch file in open_slam_ros as you specified. Since I didn't look at the launch file, I couldn't see that it launches the realsense.yaml parameter. I am sorry for that.
I also made changes to the file for the points coming from the realsense d435i camera.

When I start the mapping.launch file, I am sending the results visually in the attachment. Your work is very nice, but the 3D rendering did not work well for me compared to the other demo. Is there a point I missed?

![Screenshot from 2022-09-23 16-11-57](https://user-images.githubusercontent.com/38870269/191970905-24d111eb-fa62-46d6-af05-2728abc18c14.png)

I'm running "roslaunch realsense2_camera demo_pointcloud.launch" on the rviz image on the right.

In the rviz image on the left, I run the command "roslaunch open3d_slam_ros mapping.launch".

  What can I do to see images like the results you shared when we run the mapping.launch file?

  Thank you in advance for your reply.z
I'm running "roslaunch realsense2_camera demo_pointcloud.launch" on the rviz image on the right.

 In the rviz image on the left, I run the command "roslaunch open3d_slam_ros mapping.launch".

What can I do to see images like the results you shared when we run the mapping.launch file?
 
Thank you in advance for your reply.

**Comment by jelavice** [comment]
Hi enginksz,

it seems that you're visualizing the sparse pointcloud map. The system builds two maps, one is sparse and used for tracking and localization, while the other one is dense and used for visualization. Make sure that build_dense_map is set to true in the configuration file. Then in Rviz, you should change the visualized pointcloud topic to visualize the dense map. If you want the finer resolution, you can try to play with the voxel size of the dense map.

**Comment by jelavice** [comment]
Use two maps as suggested above. one is sparse such that tracking can run in real-time and the other one is dense which you can visualize.
